lib/runtime/allocator/freeing_bump.go

```diff
@@ -0,0 +1,586 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package allocator
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"math/bits"
+
+	"github.com/ChainSafe/gossamer/lib/runtime"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+const (
+	Aligment = 8
+
+	// each pointer is prefixed with 8 bytes, which indentifies the list
+	// index to which it belongs
+	HeaderSize = 8
+
+	// The minimum possible allocation size is chosen to be 8 bytes
+	// because in that case we would have easier time to provide the
+	// guaranteed alignment of 8
+	//
+	// The maximum possible allocation size is set to 32Mib
+	//
+	// NumOrders represents the number of orders supported, this number
+	// corresponds to the number of powers between the minimum an maximum
+	// possible allocation (2^3 ... 2^25 both ends inclusive)
+	NumOrders              uint32 = 23
+	MinPossibleAllocations uint32 = 8
```

Comment:
Can we change the way we define this to match with the doc and with the max?

```suggestion
	MinPossibleAllocations uint32 = (1 << 3)
```

Reply:
I prefer to keep it as it is, it is easier to understand values. For example, we don't like when we look at substrate docs/code and see `0x10000` to represent a `65536` (I mean, makes the reader life harder)

Reply:
Agree, I just asked it to keep some consistency since we are setting `MaxPossibleAllocations` as `(1 << 25)` representing the `2^25` instead setting the final value `33554432` but we are setting the value in `MinPossibleAllocations` instead doing the same `2^3` but obviously it's not a must just to keep consistency 👍

Reply:
yeah, I changed `MaxPossibleAllocations` to a raw uint32 value instead of a bitwise op to keep consistency